src/gitHubIssueFetcher.ts

```diff
@@ -62,29 +75,61 @@ query findByLabel($queryString:String!, $pageSize:Int, $after:String) {
             endCursor
         }
     }
+    rateLimit {
+        limit
+        cost
+        remaining
+        resetAt
+    }
 }`;
 
 export class GitHubIssueFetcher {
-    private readonly logger: mariner.Logger;
+    private readonly logger: Logger;
 
-    public constructor(logger: mariner.Logger) {
+    public constructor(logger: Logger) {
         this.logger = logger;
     }
 
     public async fetchMatchingIssues(
         token: string,
         label: string,
         repositoryIdentifiers: string[]
-    ): Promise<IssueCountAndIssues> {
-        const listOfRepos = this.createListOfRepos(repositoryIdentifiers);
-        const variables: Variables = {
-            // NOTE: See https://docs.github.com/en/github/searching-for-information-on-github/searching-issues-and-pull-requests
-            queryString: `label:\"${label}\" state:open ${listOfRepos}`,
-            pageSize: 10,
-        };
-        const issueCountAndIssues = await this.fetchAllPages(token, query, variables);
+    ): Promise<GitHubIssue[]> {
+
+        const pageSize = 100;
+        const numberOfReposPerCall = 1000;
+        const chunks = this.splitArray(repositoryIdentifiers, numberOfReposPerCall);
+
+        const edgeArray: Edge[] = [];
+        for (const chunk of chunks) {
+            const listOfRepos = this.createListOfRepos(chunk);
+            const variables: Variables = {
+                queryString: `label:\"${label}\" state:open ${listOfRepos}`,
+                pageSize,
+            };
+            const queryId = `${label}: ${chunk[0]}`;
+            const issue = await this.fetchAllPages(token, query, variables, queryId);
+            edgeArray.push(...issue.edges);
+        }
+
+        this.logger.info(`-----Fetched ${label}: ${edgeArray.length} matching issues`);
+
+        const issues = edgeArray.flatMap((edge) => {
+            return edge.node;
+        });
 
-        return issueCountAndIssues;
+        return issues;
+    }
+
+    private splitArray(repositoryIdentifiers: string[], size: number): string[][] {
+        let chunkedArray = [];
+        let i = 0;
+        for (i; i < repositoryIdentifiers.length; i += size) {
```

Comment:
I am used to seeing this as: for (let i = 0; ... rather than declaring and initializing i on its own line. I don't feel strongly about it.

Reply:
Yeah I see what you mean, I actually was using while loop instead of a for loop and when I edited it, I kept the variable out, I will edit to remove that "extra" layer.